### PR TITLE
update to use latest dp-filter

### PR DESCRIPTION
**Pull request by daiLlew**
Updated to use the latest version of dp-filter - which supports empty download filters

### CODE REVIEW

**Review by Matt-Guest — APPROVED** [review]

File: cmd/dp-dataset-exporter/main.go [hunk 1]
```diff
@@ -86,9 +89,16 @@ func main() {
...
 	observationStore := observation.NewStore(dbConnection)
 	fileStore := file.NewStore(config.AWSRegion, config.S3BucketName)
-	eventProducer := event.NewAvroProducer(kafkaProducer)
+	eventProducer := event.NewAvroProducer(kafkaProducer, schema.CSVExportedEvent)
 
-	eventHandler := event.NewExportHandler(filterStore, observationStore, fileStore, eventProducer)
+	datasetAPICli := dataset.New(config.DatasetAPIURL)
+	datasetAPICli.SetInternalToken(config.DatasetAPIAuthToken)
+
+	generateFileID := func() string {
```
Comment (Matt-Guest) [inline cmd/dp-dataset-exporter/main.go]: This can be changed to `generateFileID := uuid.NewV4().String`

File: event/filter_submitted.go [hunk 2]
```diff
@@ -2,5 +2,9 @@ package event
 
 // FilterSubmitted is the structure of each event consumed.
 type FilterSubmitted struct {
-	FilterID string `avro:"filter_output_id"`
+	FilterID      string `avro:"filter_output_id"`
```
Comment (Matt-Guest) [inline event/filter_submitted.go]: There a formatting issue here?

File: event/handler.go [hunk 3]
```diff
@@ -49,29 +63,49 @@ type ObservationStore interface {
...
+}
+
+func (handler *ExportHandler) filterJob(event *FilterSubmitted) (*CSVExported, error) {
+	log.Info("handling filter job", log.Data{"filter_id": event.FilterID})
+	filter, err := handler.filterStore.GetFilter(event.FilterID)
+	if err != nil {
+		return nil, err
+	}
 
 	csvRowReader, err := handler.observationStore.GetCSVRows(filter, nil)
```
Comment (Matt-Guest) [inline event/handler.go:103]: `csvRowReader ` does't seem to be closed?

File: event/handler_test.go [hunk 4]
```diff
@@ -319,7 +323,7 @@ func TestExportHandler_Handle(t *testing.T) {
 			Convey("The event producer is called with the filter ID.", func() {
 
 				So(len(mockedEventProducer.CSVExportedCalls()), ShouldEqual, 1)
-				So(mockedEventProducer.CSVExportedCalls()[0].FilterID, ShouldEqual, filterOutputEvent.FilterID)
+				//So(mockedEventProducer.CSVExportedCalls()[0].E., ShouldEqual, filterOutputEvent.FilterID)
```
Comment (Matt-Guest) [inline event/handler_test.go]: commented out assert

File: event/producer.go [hunk 5]
```diff
@@ -1,33 +1,39 @@
...
 	messageProducer MessageProducer
+	marshaller      Marshaller
 }
 
 // MessageProducer dependency that writes messages.
 type MessageProducer interface {
 	Output() chan []byte
 }
 
+type Marshaller interface {
```
Comment (Matt-Guest) [inline event/producer.go]: Missing go docs

File: event/producer.go [hunk 6]
```diff
@@ -1,33 +1,39 @@
...
 	}
 }
 
-// CSVExported produces a new CSV exported event for the given filter output ID.
-func (producer *AvroProducer) CSVExported(filterID, fileURL string) error {
-
-	csvExported := CSVExported{
-		FilterID: filterID,
-		FileURL:  fileURL,
+func (producer *AvroProducer) CSVExported(event *CSVExported) error {
```
Comment (Matt-Guest) [inline event/producer.go:34]: Looks like the go doc got deleted here?

File: event/handler.go [hunk 7]
```diff
@@ -82,22 +116,72 @@ func (handler *ExportHandler) Handle(event *FilterSubmitted) error {
...
-		"size":        reader.TotalBytesRead()})
-
 	// write url and file size to filter API
 	err = handler.filterStore.PutCSVData(filter.FilterID, fileURL, reader.TotalBytesRead())
 	if err != nil {
-		return err
+		return nil, errors.Wrap(err, "error while putting CSV in filter store")
+	}
+
+	return &CSVExported{FilterID: filter.FilterID, FileURL: fileURL}, nil
```
Comment (Matt-Guest) [inline event/handler.go]: Is it worth logging out the export was completed with the url?

File: config/config.go [hunk 8]
```diff
@@ -18,6 +18,8 @@ type Config struct {
 	AWSRegion                string        `envconfig:"AWS_REGION"`
 	S3BucketName             string        `envconfig:"S3_BUCKET_NAME"`
 	CSVExportedProducerTopic string        `envconfig:"CSV_EXPORTED_PRODUCER_TOPIC"`
+	DatasetAPIURL            string        `envconfig:"DATASET_API_URL"`
+	DatasetAPIAuthToken      string        `envconfig:"DATASET_API_AUTH_TOKEN"`
```
Comment (Matt-Guest) [inline config/config.go:24]: Can the readme be updated too

Comment (nshumoogum) [reply config/config.go:24]: done

File: event/handler.go (hunk 3 above)
Comment (nshumoogum) [reply event/handler.go:103]: added

File: event/producer.go (hunk 6 above)
Comment (nshumoogum) [reply event/producer.go:34]: added

**Comment by gedge** [comment]
lgtm

**Review by gedge — APPROVED** [review]

File: cmd/dp-dataset-exporter/main.go [hunk 9]
```diff
@@ -63,9 +66,14 @@ func main() {
...
 	observationStore := observation.NewStore(neo4jConnPool)
 	fileStore := file.NewStore(config.AWSRegion, config.S3BucketName)
-	eventProducer := event.NewAvroProducer(kafkaProducer)
+	eventProducer := event.NewAvroProducer(kafkaProducer, schema.CSVExportedEvent)
 
-	eventHandler := event.NewExportHandler(filterStore, observationStore, fileStore, eventProducer)
+	datasetAPICli := dataset.New(config.DatasetAPIURL)
+	datasetAPICli.SetInternalToken(config.DatasetAPIAuthToken)
+
+	generateFileID := uuid.NewV4().String()
```
Comment (CarlHembrough) [inline cmd/dp-dataset-exporter/main.go]: It seems that generating the file ID here will give the same ID to all generated downloads?

**Review by CarlHembrough — DISMISSED** [review]
The same ID is being applied to all downloads generated by the same instance of the exporter. This will mean that newly generated files will overwrite previous ones.

File: cmd/dp-dataset-exporter/main.go (hunk 9 above)
Comment (CarlHembrough) [reply cmd/dp-dataset-exporter/main.go]: It looks like the intention was to pass in an ID generating function. I have refactored to just generate a new UUID when generating the file. We can rework this later if we change the file ID's to something other than UUID's.

**Review by eldeal — APPROVED** [review]

**Pull request merged**